frontend/src/app/login/login.component.html

```diff
@@ -11,13 +11,13 @@ <h1>Connexion</h1>
       <div class="form-group">
         <label class="from-control-label" for="login">Identifiant</label>
         <input id="login" class="form-control" name="login" [(ngModel)]="credentials.login" required #loginCtrl="ngModel">
-        <div class="from-control-feedback" *ngIf="loginCtrl.control.dirty && loginCtrl.control.hasError('required')">L'identifiant est obligatoire</div>
+        <div class="invalid-feedback" *ngIf="loginCtrl.hasError('required')">L'identifiant est obligatoire</div>
```

Comment:
Nevermind, I guess it's `is-invalid`, which changes when the field is touched, that reveals the message.
These kind of directives are powerful but sometimes it's hard to follow 😅

Reply:
It's just bootstrap helping us: `.invalid-feedback` has a display of none unless its sibling form control has `.is-invalid`. So it's pure CSS, but it's helpful because it allows avoiding the repetition of the complex condition (touched or enclosing form submitted).